faulty_code.py

```
@@ -0,0 +1,86 @@
+
+def calculate_discount(price, discount):
+    """Calculate discount. Bug: negative prices allowed, floating point issues."""
+    final = price - (price * discount / 100)
+    return round(final, 2)
+    return final  # unreachable code
+
+
+class UserSession:
+    """Session manager. Bug: mutable default arg, missing init for attributes."""
+
+    active_sessions = []
```

Ruff (0.15.13)

[warning] Mutable default value for class attribute

(RUF012)